### PR TITLE
Transaction feature

### PR DESCRIPTION
**Pull request by josefwijaya**

### CODE REVIEW

File: lib/intergallactic/transaction.js [hunk 1]
```diff
@@ -24,7 +33,7 @@ const config = require('../config/constant'),
  * };
  * const opt = { txnType: 1 // required }
  * const newTxn = new igc.Transaction(txn, opt);
- * newTxn.signNSend(privateKey)
+ * newTxn.signNBroadcast(privKey)
```
Comment (b00f) [inline lib/intergallactic/transaction.js]: I think better to broadcast a signed tx. Sign the broadcast. More secure.

File: lib/intergallactic/transaction.js [hunk 2]
```diff
@@ -34,11 +43,11 @@ class Transaction {
       throw new Error('Instatiate transaction require type value to decide whether send or call transaction.')
```
Comment (ToniyaSundaram) [inline lib/intergallactic/transaction.js]: *Instantiate

File: lib/intergallactic/transaction.js (hunk 1 above)
Comment (josefwijaya) [reply lib/intergallactic/transaction.js]: you mean to change the naming from signNBroadcast to broadcastSignedTx?

Comment (b00f) [reply lib/intergallactic/transaction.js]: It's better to not pass private_key between objects. You can have a `signer` object to just sign transactions. This object can hold the private_key.

Comment (josefwijaya) [reply lib/intergallactic/transaction.js]: I agree.. In this case my `signer` would be the `signByPrivKey` function.

**Pull request merged**